lib/libutils/ext/include/util.h

```diff
@@ -140,6 +140,21 @@
 #define GENMASK_64(h, l) \
 	(((~UINT64_C(0)) << (l)) & (~UINT64_C(0) >> (64 - 1 - (h))))
 
+/* Get and set bit fields  */
+#define GET_FIELD_U32(reg, mask) \
+	(((reg) & UINT32_C(mask)) / ((mask) & ~((mask) << 1)))
```

Comment:
Requiring `mask` to be a constant instead may make this macro somewhat less useful.
The other uses of `mask` should also be made into an unsigned 32-bit integer. If we don't care about assembly it's probably easier to rewrite it as (with the bonus that `mask` is only evaluated once):
``` C
#define GET_FIELD_U32(reg, mask) \
        (__extension__({ \
                uint32_t _mask  = (mask); \
                \
        ￼       ((reg) & _mask) / (_mask & ~SHIFT_U32(_mask, 1); \
        }))